migration/tasklist-migration/src/test/java/io/camunda/migration/tasklist/ElasticsearchMigrationRunnerIT.java

```diff
@@ -0,0 +1,245 @@
+
+    r =
+        new CreateIndexRequest.Builder()
+            .index(IndexUtil.getTargetIndexName(properties))
+            .withJson(new StringReader(TestData.operateIndex()))
+            .build();
+
+    esClient.indices().create(r).acknowledged();
+  }
+
+  @BeforeEach
+  public void setup() throws IOException {
+    migrator = new MigrationRunner(properties);
+    esClient.deleteByQuery(
+        DeleteByQueryRequest.of(
+            d ->
+                d.index(IndexUtil.getSourceIndexName(properties))
+                    .conflicts(Conflicts.Proceed)
+                    .query(q -> q.matchAll(m -> m))));
+    esClient.deleteByQuery(
+        DeleteByQueryRequest.of(
+            d ->
+                d.index(IndexUtil.getTargetIndexName(properties))
+                    .conflicts(Conflicts.Proceed)
+                    .query(q -> q.matchAll(m -> m))));
+  }
+
+  @Test
+  public void shouldMigrateSuccessfully() throws IOException {
+    // when
+    createEntity(
+        new ProcessEntity()
+            .setId("1")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithForm())
+            .setBpmnProcessId("formProcess"),
+        true);
+    createEntity(
+        new ProcessEntity()
+            .setId("2")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithoutForm())
+            .setBpmnProcessId("formProcess"),
+        true);
+    awaitRecordsArePresent();
+    // then
+    migrator.run();
+
+    // verify
+    assertTargetRecordsAreMigrated();
+    assertSourceRecordsAreDeleted();
+  }
+
+  @Test
+  public void shouldMigrateSuccessfullyWithMultipleRounds() throws IOException {
+    // when
+    for (int i = 0; i < 20; i++) {
+      createEntity(
+          new ProcessEntity()
+              .setId(String.valueOf(i))
+              .setVersion(1)
+              .setBpmnXml(TestData.xmlWithForm())
+              .setBpmnProcessId("formProcess"),
+          true);
+    }
+    awaitRecordsArePresent();
+    // then
+    migrator.run();
+
+    // verify
+    final var records =
+        readRecords(
+            io.camunda.webapps.schema.entities.operate.ProcessEntity.class,
+            IndexUtil.getTargetIndexName(properties));
+    assertThat(records.size()).isEqualTo(20);
+    assertThat(records.stream().noneMatch(r -> r.getIsPublic().equals(Boolean.FALSE))).isTrue();
+    assertThat(records.stream().noneMatch(r -> r.getFormId() == null)).isTrue();
+    assertSourceRecordsAreDeleted();
+  }
+
+  @Test
+  public void migrationShouldFail() throws IOException {
+    // when
+    createEntity(
+        new ProcessEntity()
+            .setId("1")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithForm())
+            .setBpmnProcessId("formProcess"),
+        true);
+    createEntity(
+        new ProcessEntity()
+            .setId("2")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithoutForm())
+            .setBpmnProcessId("formProcess"),
+        false);
```

Comment:
🔧 Reading this it is not clear to me what these flags mean. 

I would propose that we have two methods one `createEntity` and one `createEntityWithDuplicate`